### PR TITLE
Support metric system measurements (#4)

### PR DESCRIPTION
**Pull request by prattheev**
This PR consists the fix for [#4](https://github.com/hackclub/finder/issues/4).

## Changes
 - Makes use of user's location to identify country to predict unit of measure.
 - Added Toggle button for unit of measure.
 - Added fuse support for metric systems.

@ifvictr Sorry about the delay, something came up :relieved: 
Let me know if it is okay! :smiley:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,3 @@
 .c9/
-.git/
```
Comment (ifvictr) [inline .gitignore]: Could you keep those lines?

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,3 @@
...
 node_modules/
 npm-debug.log
-package-lock.json
```
Comment (ifvictr) [inline .gitignore]: Could you keep those lines?

File: assets/js/main.js [hunk 3]
```diff
@@ -3,6 +3,10 @@ if(!navigator.geolocation) {
     console.warn("Couldn't use geolocation");
 }
 
+// Global Variable
+var defaultUnit = "Imperial";
```
Comment (ifvictr) [inline assets/js/main.js]: Could you use `const` or `let` instead of `var` for everything?

File: assets/js/main.js [hunk 4]
```diff
@@ -162,8 +168,45 @@ $(() => {
...
         $("[data-type='all']").toggleClass("is-hidden");
         $("[data-output='location']").toggleClass("is-hidden");
         updateResults(clubs, coords, this.checked);
     });
-});
\ No newline at end of file
+    $("#toggle-unit").on("change", function() {
+        var el = $("[for='toggle-unit']");
+        var currentUnit = el.html();
+        if(currentUnit == "Imperial"){
```
Comment (ifvictr) [inline assets/js/main.js]: Could you follow the convention? There are spaces between the closing parentheses and opening braces, and newlines after closing braces.

File: assets/js/util.js [hunk 5]
```diff
@@ -21,4 +21,5 @@ const isSuccess = (url, callback) => {
     });
 };
 
-const toMiles = kilometers => kilometers * 0.000621371;
\ No newline at end of file
+const toMiles = metres => metres * 0.000621371;
+const toKilometres = metres => metres * 0.001;
```
Comment (ifvictr) [inline assets/js/util.js]: Could you spell it as `meters`?

File: index.html [hunk 6]
```diff
@@ -48,13 +48,19 @@
...
                             <div class="level-item" data-type="nearby">
-                                <p><output data-output="radius">5</output> mile radius</p>
+                                <p><output data-output="radius">5 mile</output> radius</p>
                             </div>
                             <div class="level-item" data-type="nearby">
                                 <div class="field">
                                     <input class="slider is-circle is-medium" name="radius" type="range" min="1" max="100" value="5">
                                 </div>
                             </div>
+                            <div class="level-item" data-type="baseUnit">
```
Comment (ifvictr) [inline index.html]: Could you separate words with dashes instead of doing camel-case?

File: assets/js/main.js [hunk 7]
```diff
@@ -121,6 +126,7 @@ $(() => {
                     .done(data => {
                         if(data.results.length > 0) {
                             const result = data.results[0];
+                            updateUnit(data);
```
Comment (ifvictr) [inline assets/js/main.js]: Could you move it so it isn't in the middle of two variable declarations?

File: assets/js/main.js [hunk 8]
```diff
@@ -162,8 +168,45 @@ $(() => {
...
+            defaultDistance = "mile";
+        }
+        el.html(defaultUnit);
+        const val = $("input[type='range']").val();
+        $("[data-output='radius']").html(val + " " + defaultDistance);
+        updateResults(clubs, coords);
+    });
+});
+
+const getDistance = distance => defaultUnit == "Imperial" ? toMiles(distance) : toKilometres(distance);
```
Comment (ifvictr) [inline assets/js/main.js]: Could you move this to `util.js`?

File: assets/js/main.js [hunk 9]
```diff
@@ -162,8 +168,45 @@ $(() => {
...
+
+function updateUnit(res){
```
Comment (ifvictr) [inline assets/js/main.js]: Could you move this to `util.js`?

**Review by ifvictr — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (prattheev) [reply .gitignore]: That's not best practices!

File: .gitignore (hunk 2 above)
Comment (prattheev) [reply .gitignore]: Okay!

**Comment by prattheev** [comment]
@ifvictr Fixed them!

File: assets/js/main.js (hunk 7 above)
Comment (prattheev) [reply assets/js/main.js]: Done!

**Pull request merged**